src/api/base-http-client/base-http-client.ts

```diff
@@ -10,34 +12,39 @@ export abstract class BaseHttpClient implements HttpClient {
     this.baseUrl = baseUrl;
   }
 
-  async request<T = any, D = any>(
-    { route, ...rest }: Config<T>,
+  async request<D = any>(
+    { route, params, ...rest }: Config,
     isCallingBackgroundScript: boolean = true
   ): Promise<Response<D>> {
     const url = `${this.baseUrl}${route}`;
+    const config = {
+      ...(params && {
```

Comment:
nit: hmm.. what's happening here :?

![image](https://user-images.githubusercontent.com/43485369/169241803-3f3113b7-6075-4e69-a2ff-6d13f739b06e.png)

Reply:
It is optional spreading of the `searchParams` key if `params` exists. Maybe a ternary would've been easier to read?